biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -183,7 +201,11 @@ private Repository readRepository() throws Exception {
 				}
 				return fetch(a, 2, 1000L).map(tag -> {
 					ResourceBuilder rb = new ResourceBuilder();
-					rb.addFile(tag.getFile(), a.uri);
+					File file = tag.getFile();
```

Comment:
I would have assumed we would have configured the promise to have performed the unpack operation so we don't have to do it everywhere else.

Reply:
The unpacking should be done as a map in the fetch result:

```java
return fetch(a, 2, 1000L).map(tag -> Artifact.FORMAT_PACKED.equals(a.format) ? 
  handleDownloadResult(true, null, tag.getFile()) : tag.getFile())
.map(file -> {
  ResourceBuilder rb = new ResourceBuilder();
  rb.addFile(file, a.uri);
  ...
})
```